### PR TITLE
feat(schema): add `original_status` and `original_result` fields to all related devops tables

### LINKED ISSUE #6139: [Feature][Schema] Add `original_status` and `original_result` to all CICD entities
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-devlake/issues?q=is%3Aissue) and found no similar feature requirement.

### Use case

DevLake uses hardcoded rules to convert a tool's original statuses or results to domain layer `status` and `result`. This is because most of the tools' statuses and results are quite stable and cannot be customized.

However, there might be a case that users are not satisfied with the mapping rules hardcoded in DevLake. And adding `original_status` and `original_result` to CICD entities can address this problem. Users can customize the SQL with these fields, just like the original_type and original_status in table.issues.

### Description

There are a few todo items:
- [ ] Add `original_status` and `original_result` to all CICD entities: cicd_pipelines, cicd_tasks, cicd_deployments and cicd_deployment_commits
- [ ] Use each CI tool's original status as the value of `original_status`
- [ ] Use each CI tool's original result as the value of `original_result`. If the CI tool doesn't have a result attribute, leave it empty.
- [ ] Update the domain layer schema doc and diagram

### Related issues

_No response_

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by d4x1**
### ⚠️ Pre Checklist

> Please complete _ALL_ items in this checklist, and remove before submitting

- [x] I have read through the [Contributing Documentation](https://devlake.apache.org/community/).
- [x] I have added relevant tests.
- [x] I have added relevant documentation.
- [x] I will add labels to the PR, such as `pr-type/bug-fix`, `pr-type/feature-development`, etc.

### Summary
What does this PR do?
1. add original_status and original_result fields to all related devops tables
2. add migration script
3. fulfill these fields in all cicd plugins

### Does this close any open issues?
Closes  #6139 

### Screenshots
Include any relevant screenshots here.
#### cicd_tasks
image

#### cicd_pipelines
image

#### cicd_deployments
image

#### cicd_deployment_commits
image

### Other Information
Any other information that is important to this PR.

### CODE REVIEW

**Comment by d4x1** [comment]
The documents should be updated.

File: backend/core/models/migrationscripts/archived/cicd_deployment.go [hunk 1]
```diff
@@ -28,6 +28,8 @@ type CicdDeploymentCommit struct {
 	Name                          string `gorm:"type:varchar(255)"`
 	Result                        string `gorm:"type:varchar(100)"`
 	Status                        string `gorm:"type:varchar(100)"`
+	OriginalStatus                string `gorm:"type:varchar(100)"`
```
Comment (klesh) [inline backend/core/models/migrationscripts/archived/cicd_deployment.go:31]: No, do NOT modify the **archived** models

Comment (d4x1) [reply backend/core/models/migrationscripts/archived/cicd_deployment.go]: @klesh Got it! And I fix it.

File: backend/core/models/migrationscripts/20231116_add_original_status_and_result_to_devops_tables.go [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
+import (
+	"github.com/apache/incubator-devlake/core/context"
+	"github.com/apache/incubator-devlake/core/errors"
+	"github.com/apache/incubator-devlake/core/plugin"
+)
+
+var _ plugin.MigrationScript = (*addOriginalStatusAndResultToDevOpsTables)(nil)
+
+type cicdDeployment2023116 struct {
```
Comment (abeizn) [inline backend/core/models/migrationscripts/20231116_add_original_status_and_result_to_devops_tables.go:28]: 20231116?

Comment (d4x1) [reply backend/core/models/migrationscripts/20231116_add_original_status_and_result_to_devops_tables.go]: Updated !

**Review by abeizn — APPROVED** [review]

**Pull request merged**